### PR TITLE
docs: add a "Development Best Practices" section

### LINKED ISSUE #612: Create README with dev best practices
From @carolinemodic:

> start google doc (or github readme?) with all our best practices around coding (i.e. use zod schemas, use readonly properties, whatever we decide with exceptions/control flow, etc.) 

Probably include description of votingworks/vxsuite#610

### PR DESCRIPTION
**Pull request by eventualbuddha**
Closes #612

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,6 +89,114 @@ To add a workspace package `foo` as a dependency, do this:
...
+**Example: Use `const` instead of `let` (and never use `var`)**
+
+This should be enforced by eslint when a variable is never reassigned. This doesn't mean you should _never_ use `let`, just that a `const` version might be better. If it's better to use `let`, that's fine. There's almost no reason to use `var`, though.
+
+**Example: Update objects and arrays with rest values**
+
+Rather than assigning to an object property, consider building a new object:
+
+```ts
+// toggle test mode by mutating `settings` (not preferred)
```
Comment (benadida) [inline README.md:164]: make the not-preferred solution more clearly not preferred.

```suggestion
// NOT PREFERRED: toggle test mode by mutating `settings`
```

File: README.md [hunk 2]
```diff
@@ -89,6 +89,114 @@ To add a workspace package `foo` as a dependency, do this:
...
+settings.testMode = !settings.testMode
+
+// toggle test mode by creating a new object (preferred)
```
Comment (benadida) [inline README.md:167]:
```suggestion
// PREFERRED: toggle test mode by creating a new object
```

**Review by benadida — APPROVED** [review]

**Review by carolinemodic — APPROVED** [review]
annayay

**Comment by carolinemodic** [comment]
One thought of something else to add which you are probably most familiar with is using debug within our code, how we should name modules for that, how much we should put debug lines, etc.

**Comment by eventualbuddha** [comment]
@carolinemodic thanks for the suggestion, added.

**Pull request merged**